vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceLinkManager.java

```diff
@@ -109,5 +109,21 @@ protected void deleteLinkRepo(ISpaceLink link) {
     public List<ISpaceLinkDisplay> getSpaceLinkForGivenOrNullSpaceStatus(String spaceId, SpaceStatus spaceStatus){
         return new ArrayList<>(spaceLinkDisplayRepo.findSpaceLinksForGivenOrNullSpaceStatus(spaceId,spaceStatus));
     }
+    
+    @Override
+    public void deleteSpaceLinksWithSourceAsNull(){
+        List<ISpaceLink> spaceLinks = spaceLinkRepo.findBySourceSpaceIsNull();
+        //need to delete all space link displays before deleting spacelink
+        for(ISpaceLink spaceLink : spaceLinks) {
+            spaceLinkDisplayRepo.deleteByLink(spaceLink);
+        }
```

Comment:
I think you can create a repository method `deleteByLinkIn` and then give it the whole list, can't you?  To avoid multiple delete operations.

Reply:
Done